### PR TITLE
Add PHP CodeSniffer to travis, check only modified files

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Travis now runs PHP CodeSniffer on the modified files in order to enforce PSR-2 rules (except the namespace on non-namespaced legacy files).<br>Also, the PHP linter is now executed on modified files only.
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  |  Fork this branch, introduce problems in your PHP files, commit, see the build fail 💥<br>Commit a fix, see the build pass ✅

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/7979)

### CODE REVIEW

**Review by aleeks — DISMISSED** [review]
Looks like ok, as we said, we need to clean all file before merge it!

**Comment by eternoendless** [comment]
Here's the PR for the cleanup: https://github.com/PrestaShop/PrestaShop/pull/7997

**Comment by Soullivaneuh** [comment]
You can also configure this one: http://cs.sensiolabs.org/

It's more a fixer than a sniffer with additional functionalities and rules which may interest you.

You can also run it on Travis with the `--dry-run` option to have the diff and a return code if code differs.

I could provide a PR if you are interested.

**Comment by eternoendless** [comment]
Hi @Soullivaneuh

Thanks for this suggestion, I'll look into it and see how it can be integrated.

**Comment by Quetzacoalt91** [comment]
Can you please rebase in order to fix the conflicts in  composer.lock please ?

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Seen with @eternoendless, we also want to check if there are other solutions for the standards on each PR. That's why this one is not merged yet.

File: tests/phpcs-ignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+app/SymfonyRequirements.php
+vendor/*
+tools/*
+modules/*
```
Comment (soullivaneuh) [inline tests/phpcs-ignore:4]: A new line at end of file may be added.

**Review by soullivaneuh — CHANGES_REQUESTED** [review]

**Comment by Quetzacoalt91** [comment]
@eternoendless, if I remember well, you wanted to go back on the standard verification, and only check the syntax on modified files.

Could you please update your PR for this purpose?

**Comment by Quetzacoalt91** [comment]
This PR is now a duplicate of https://github.com/PrestaShop/PrestaShop/pull/9468

I'm closing it. Feel free to reopen it if something still needs to be brought to the repository.

**Pull request closed without merging**